ListResponse does not validate correctly if schemas[0] is not primary schema

[RFC 7643, Section 3](https://datatracker.ietf.org/doc/html/rfc7643#section-3) states

> "Schemas" Attribute
      [...]
      Value order is not specified and MUST NOT impact behavior.

I have added this test to `test_list_response.py`:

```python
def test_list_response_schema_ordering():
    """Test that the "schemas" attribute order does not impact behavior
    https://datatracker.ietf.org/doc/html/rfc7643#section-3"""

    payload = {
        "totalResults": 1,
        "Resources": [
            {
                "schemas": [
                    "urn:ietf:params:scim:schemas:extension:enterprise:2.0:User",
                    "urn:ietf:params:scim:schemas:core:2.0:User",
                ],
                "userName": "bjensen@example.com",
            }
        ]
    }
    ListResponse.of(User[EnterpriseUser], Group).model_validate(payload)
```

This fails with

> E       pydantic_core._pydantic_core.ValidationError: 1 validation error for ListResponse[Annotated[Union[Annotated[User[EnterpriseUser], Tag], Annotated[Group, Tag]], Discriminator]]
E       resources.0
E         Input tag 'urn:ietf:params:scim:schemas:extension:enterprise:2.0:User' found using get_schema_from_payload() does not match any of the expected tags: 'urn:ietf:params:scim:schemas:core:2.0:User', 'urn:ietf:params:scim:schemas:core:2.0:Group' [type=union_tag_invalid, input_value={'schemas': ['urn:ietf:pa...: 'bjensen@example.com'}, input_type=dict]
E           For further information visit https://errors.pydantic.dev/2.8/v/union_tag_invalid

The cause is that `scim2_models.rfc7643.resource.tagged_resource_union.get_schema_from_payload` just [gets the first entry from `payload["schemas"]`](https://github.com/yaal-coop/scim2-models/blob/0.1.11/scim2_models/rfc7643/resource.py#L348).

--- a/doc/changelog.rst
+++ b/doc/changelog.rst
@@ -4,6 +4,7 @@ Changelog
 Fixed
 ^^^^^
 - Additional bugfixes about attribute case sensitivity #45
+- :class:`~scim2_models.ListResponse` tolerate any schema order #50
 
 [0.1.11] - 2024-07-02
 ---------------------

--- a/scim2_models/rfc7643/resource.py
+++ b/scim2_models/rfc7643/resource.py
@@ -353,10 +353,19 @@ def tagged_resource_union(resource_types: Resource):
         return resource_types
 
     def get_schema_from_payload(payload: Any):
-        try:
-            return payload["schemas"][0]
-        except KeyError:
+        if not payload:
             return None
+
+        resource_types_schemas = [
+            resource_type.model_fields["schemas"].default[0]
+            for resource_type in resource_types
+        ]
+        common_schemas = [
+            schema
+            for schema in payload.get("schemas")
+            if schema in resource_types_schemas
+        ]
+        return common_schemas[0] if common_schemas else None
 
     def get_tag(resource_type: Type):
         return Tag(resource_type.model_fields["schemas"].default[0])

--- a/tests/test_list_response.py
+++ b/tests/test_list_response.py
@@ -185,3 +185,22 @@ def test_zero_results():
         ListResponse.of(User).model_validate(
             payload, scim_ctx=Context.RESOURCE_QUERY_RESPONSE
         )
+
+
+def test_list_response_schema_ordering():
+    """Test that the "schemas" attribute order does not impact behavior
+    https://datatracker.ietf.org/doc/html/rfc7643#section-3"""
+
+    payload = {
+        "totalResults": 1,
+        "Resources": [
+            {
+                "schemas": [
+                    "urn:ietf:params:scim:schemas:extension:enterprise:2.0:User",
+                    "urn:ietf:params:scim:schemas:core:2.0:User",
+                ],
+                "userName": "bjensen@example.com",
+            }
+        ],
+    }
+    ListResponse.of(User[EnterpriseUser], Group).model_validate(payload)

